webapp/crashstats/signature/tests/test_views.py

```diff
@@ -2,506 +2,524 @@
             url,
             {
-                "signature": DUMB_SIGNATURE,
-                "product": "WaterWolf",
+                "signature": TEST_SIGNATURE,
+                "product": "Firefox",
+                "date": [f">={start_date}", f"<{end_date}"],
                 "_columns": ["build_id", "platform"],
             },
         )
         assert response.status_code == 200
         assert 'table id="reports-list"' in smart_str(response.content)
         # The build and platform appear
-        assert "888981" in smart_str(response.content)
-        assert "Linux" in smart_str(response.content)
+        assert str(crash1["build"]) in smart_str(response.content)
+        assert crash1["os_name"] in smart_str(response.content)
         # The crash id is always shown
-        assert "f74a5763-3270-4151-9c49-853710220208" in smart_str(response.content)
+        assert crash1["uuid"] in smart_str(response.content)
         # The version and date do not appear
-        assert "1.0" not in smart_str(response.content)
-        assert "2017" not in smart_str(response.content)
+        assert crash1["version"] not in smart_str(response.content)
+        assert crash1_date_processed not in smart_str(response.content)
+
+    def test_missing_parameters(self, client, db, es_helper):
+        url = reverse("signature:signature_reports")
 
         # Test missing parameter.
-        response = self.client.get(url)
+        response = client.get(url)
         assert response.status_code == 400
 
-        response = self.client.get(url, {"signature": ""})
+        response = client.get(url, {"signature": ""})
         assert response.status_code == 400
 
-    def test_parameters(self):
+    def test_parameter_parsing(self, client, db, es_helper):
+        calls = []
+
         def mocked_supersearch_get(**params):
-            # Verify that all expected parameters are in the URL.
-            assert "product" in params
-            assert "WaterWolf" in params["product"]
-            assert "NightTrain" in params["product"]
+            calls.append(params)
+            return {"hits": [], "facets": "", "total": 0}
 
-            assert "address" in params
+        with mock.patch(
+            "crashstats.supersearch.models.SuperSearchUnredacted.get_implementation"
+        ) as mocked_get_implementation:
+            mocked_get_implementation.return_value.get.side_effect = (
+                mocked_supersearch_get
+            )
+
+            date = utc_now().strftime("%Y-%m-%d")
+
+            url = reverse("signature:signature_reports")
+            response = client.get(
+                url,
+                {
+                    "signature": TEST_SIGNATURE,
+                    "product": ["Firefox", "Thunderbird"],
+                    "address": ["0x0", "0xa"],
+                    "reason": ["^hello", "$thanks"],
+                    "date": [f">={date}"],
+                    "java_stack_trace": "Exception",
```

Comment:
This is a contrived impossible scenario since we'll never have a Firefox or Thunderbird crash report with `java_stack_trace`.